core/lib/common/worker.js

```diff
@@ -106,6 +106,16 @@ module.exports = class Worker {
 			this.workerPort = '22';
 			this.sshPrefix = `host ${this.uuid} `;
 		}
+		if (this.url.includes(`bm.balena-dev.com`)) {
+			// worker is a testbot connected to balena cloud - we ssh into it via the vpn
+			this.uuid = this.url.match(
+				/https:\/\/([^\.]+)\./,
+			)[1];
+			this.workerHost = `ssh.devices.bm.balena-dev.com`;
+			this.workerUser = this.username;
+			this.workerPort = '222';
+			this.sshPrefix = `host ${this.uuid} `;
+		}
```

Comment:
Could we get these supported as values in the config instead? Would be much easier to scale in the future and again hardcoding. What if the BM machine URL change or there is a new BM instance in place. Changing a config for the user is much easier than changing code in core. 

Can discuss how this can look towards user on the config side

Reply:
will require adding a new argument to the worker class, and then modifying the test suite to pass the config (or at least part of it) to the constructor

I think its a better idea also though yeah

Reply:
@vipulgupta2048 I pushed a new commit to achieve this - using defaults which mean that if no config is provided, then it will use the old values - should maintain backwards compatibility with the tests